### PR TITLE
Migrate click behavior column computation to MLv2

### LINKED ISSUE #37342: [MLv2] [FE] [Blocked] Use a MLv2 function to compute the list of available columns in click behaviors
blocked by https://github.com/metabase/metabase/issues/37501

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/37342

Blocked by https://github.com/metabase/metabase/issues/37501 and depends on https://github.com/metabase/metabase/pull/37648

### CODE REVIEW

**Comment by uladzimirdev** [comment]
There is one test broken, but most likely the fix should be on BE side

https://github.com/metabase/metabase/pull/37648#issuecomment-1891629756

File: e2e/test/scenarios/dashboard-cards/dashboard-drill.cy.spec.js [hunk 1]
```diff
@@ -179,10 +179,10 @@ describe("scenarios > dashboard > dashboard drill", () => {
 
     modal().findByText("Orders").click();
 
-    sidebar().findByText("Orders → User ID").click();
+    sidebar().findByText("User ID").click();
```
Comment (uladzimirdev) [inline e2e/test/scenarios/dashboard-cards/dashboard-drill.cy.spec.js:182]: naming in MLv2 has changed

File: frontend/src/metabase-lib/parameters/utils/click-behavior.ts [hunk 2]
```diff
@@ -124,75 +127,138 @@ export function getDataFromClicked({
...
 
     return {
       id,
       target,
-      name: o.displayName({ includeTable: true }),
+      name: Lib.displayInfo(query, stageIndex, targetColumn).longDisplayName,
       sourceFilters: {
-        column: column =>
+        column: (sourceColumn, sourceQuestion) => {
+          if (!sourceQuestion) {
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/parameters/utils/click-behavior.ts:165]: we pass `question` as `query` from the closure is becoming outdated

File: frontend/src/metabase-lib/column_types.ts [hunk 3]
```diff
@@ -38,3 +39,10 @@ export const isTime: TypeFn = TYPES.time_QMARK_;
 export const isTitle: TypeFn = TYPES.title_QMARK_;
 export const isURL: TypeFn = TYPES.URL_QMARK_;
 export const isZipCode: TypeFn = TYPES.zip_code_QMARK_;
+
+export function isCompatibleType(
+  column1: ColumnMetadata | DatasetColumn,
```
Comment (ranquild) [inline frontend/src/metabase-lib/column_types.ts:44]: @uladzimirdev it's only `ColumnMetadata` now

File: frontend/src/metabase-lib/parameters/utils/click-behavior.ts [hunk 4]
```diff
@@ -41,7 +44,7 @@ interface Target {
 }
 
 interface SourceFilters {
-  column: (column: DatasetColumn) => boolean;
+  column: (column: DatasetColumn, question?: Question) => boolean;
```
Comment (ranquild) [inline frontend/src/metabase-lib/parameters/utils/click-behavior.ts:47]: Can we make it required?

File: frontend/src/metabase-lib/parameters/utils/click-behavior.ts (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase-lib/parameters/utils/click-behavior.ts:165]: If `sourceQuestion` is required this won't be needed

**Comment by ranquild** [comment]
@uladzimirdev left some comments

File: frontend/src/metabase-lib/parameters/utils/click-behavior.ts (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/parameters/utils/click-behavior.ts]: https://github.com/metabase/metabase/pull/37438/commits/aff38ff66570bf46a3851a6e926083b792893c49

File: frontend/src/metabase-lib/column_types.ts (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/column_types.ts]: [63f9ae6](https://github.com/metabase/metabase/pull/37438/commits/63f9ae6b4b6112a1a11544c44874d77dfeac0677)

**Comment by ranquild** [comment]
@uladzimirdev looks good! Please change the target branch to the PR where `fromLegacyColumn` is added. Otherwise LGTM. Feel free to approve since I cannot :)

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**